### PR TITLE
domain/encrypt: extract logic and cover tests

### PR DESCRIPTION
**Pull request by pleshevskiy**
- [x] domain/encrypt: extract logic from cli
- [x] domain/encrypt: covert with unit tests
- [x] cli/encrypt: use domain logic
- [ ] ~~optimize encryption for tests.~~

### CODE REVIEW

File: dexios/src/domain/encrypt.rs [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+            },
+            hashing_algorithm: HashingAlgorithm::Blake3Balloon(5),
+        };
+
+        match execute(req) {
+            Ok(_) => {
+                assert_eq!(
+                    output_content,
+                    vec![
+                        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
```
Comment (pleshevskiy) [inline dexios/src/domain/encrypt.rs:295]: @brxken128 Why do we have to fill content with zeros when we detach the header?

**Comment by brxken128** [comment]
All looks great, I'll merge this! :)

File: dexios/src/domain/encrypt.rs (hunk 1 above)
Comment (pleshevskiy) [reply dexios/src/domain/encrypt.rs:292]: @brxken128 I don't understand)

Comment (pleshevskiy) [reply dexios/src/domain/encrypt.rs:292]: we just skip header_size (and fill content with zeros) when we detach the headers

**Pull request merged**